huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_code_table_values.go

```diff
@@ -0,0 +1,369 @@
+		"code_table_field_values": buildCreateCodeTableFieldValuesParams(d),
+	}
+	return map[string]interface{}{
+		"to_add": params,
+	}
+}
+
+func buildCreateCodeTableFieldValuesParams(d *schema.ResourceData) []map[string]interface{} {
+	rawValues := d.Get("values").([]interface{})
+	values := make([]map[string]interface{}, len(rawValues))
+	for i, rawValue := range rawValues {
+		valueMap := rawValue.(map[string]interface{})
+		value := map[string]interface{}{
+			"fd_id":    d.Get("field_id"),
+			"fd_value": valueMap["value"].(string),
+			"ordinal":  i + 1,
+		}
+		values[i] = value
+	}
+	return values
+}
+
+func resourceArchitectureCodeTableValuesRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	workspaceID := d.Get("workspace_id").(string)
+
+	var mErr *multierror.Error
+	getCodeTableValuesProduct := "dataarts"
+
+	client, err := cfg.NewServiceClient(getCodeTableValuesProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Studio client: %s", err)
+	}
+
+	getCodeTableFieldValuesPath, getCodeTableFieldValuesOpt := buildCodeTableValuesPathAndOpt(client, d)
+	getCodeTableFieldValuesResp, err := client.Request("GET", getCodeTableFieldValuesPath, &getCodeTableFieldValuesOpt)
+	if err != nil {
+		return diag.Errorf("error retrieving DataArts Architecture code table field values: %s", err)
+	}
+
+	getCodeTableFieldValuesRespBody, err := utils.FlattenResponse(getCodeTableFieldValuesResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	fieldName := d.Get("field_name").(string)
+	values, err := flattenCodeTableFieldValues(getCodeTableFieldValuesPath, client, getCodeTableFieldValuesOpt, fieldName)
+	if err != nil {
+		return diag.Errorf("error retrieving DataArts Architecture code table field values: %s", err)
+	}
+
+	if len(values) == 0 {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "error retrieving DataArts Architecture code table field values")
+	}
+
+	findFieldExpr := fmt.Sprintf("data.value.records[?name_ch=='%s'] | [0]", fieldName)
+	curJson := utils.PathSearch(findFieldExpr, getCodeTableFieldValuesRespBody, nil)
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("workspace_id", workspaceID),
+		d.Set("table_id", utils.PathSearch("code_table_id", curJson, nil)),
+		d.Set("field_id", utils.PathSearch("id", curJson, nil)),
+		d.Set("field_ordinal", utils.PathSearch("ordinal", curJson, nil)),
+		d.Set("field_name", utils.PathSearch("name_ch", curJson, nil)),
+		d.Set("field_code", utils.PathSearch("name_en", curJson, nil)),
+		d.Set("field_type", utils.PathSearch("data_type", curJson, nil)),
+		d.Set("values", values),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func flattenCodeTableFieldValues(getCodeTableFieldValuesPath string, client *golangsdk.ServiceClient,
+	getCodeTableFieldValuesOpt golangsdk.RequestOpts, fieldName string) ([]interface{}, error) {
```

Comment:
suggest to make the parameters with same type together

Reply:
fixed.